File: 27. Remove Element/py/removeElement.py
# LeetCode Problem: 27. Remove Element
# <https://leetcode.com/problems/remove-element/description/>


class Solution:
    def removeElement(self, nums: list[int], val: int) -> int:
        index = 0
        while index < len(nums):
            if nums[index] == val:
                del nums[index]
            else:
                index += 1
        return len(nums)

File: 27. Remove Element/cs/RemoveElement/RemoveElement.cs
// LeetCode Problem: 27. Remove Element
// https://leetcode.com/problems/remove-element/description/

namespace LeetCode.RemoveElement;

public class Solution
{
    public int RemoveElement(int[] nums, int val)
    {
        int index = 0;
        for (int i = 0; i < nums.Length; i++)
        {
            if (nums[i] != val)
                nums[index++] = nums[i];
        }
        return index;
    }
}